app/client/src/components/RoleGuard.tsx

```diff
@@ -0,0 +1,14 @@
+import router from 'next/router';
+
+export interface RoleGuardProps {
```

Comment:
Maybe add a `redirect` string param so that can be customized?  If it's missing, then we don't redirect at all.  Some things we may just not display to users, but overall the page stays the same regardless of moderator vs user.